hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/magic/MagicS3GuardCommitter.java

```diff
@@ -264,9 +326,14 @@ public void abortTask(TaskAttemptContext context) throws IOException {
     try (DurationInfo d = new DurationInfo(LOG,
         "Abort task %s", context.getTaskAttemptID());
         CommitContext commitContext = initiateTaskOperation(context)) {
-      getCommitOperations().abortAllSinglePendingCommits(attemptPath,
-          commitContext,
-          true);
+      if (isTrackMagicCommitsInMemoryEnabled(context.getConfiguration())) {
+        List<SinglePendingCommit> pendingCommits = loadPendingCommitsFromMemory(context);
+        for (SinglePendingCommit singleCommit : pendingCommits) {
+          commitContext.abortSingleCommit(singleCommit);
+        }
+      } else {
+        getCommitOperations().abortAllSinglePendingCommits(attemptPath, commitContext, true);
```

Comment:
hmmm. this is trouble here as abortTask may be called from the job process rather than the task attempt. For example, TA considered failed; job instances calls abort. we can do that today because a list will find all pending uploads. with in memory, there's no longer that ability

Reply:
AFIK, Spark calls abortTask from the same process (executor), When the job fails, The abortJob operation is called which basically lists all the pending uploads and aborts it as mentioned in the comment [here](https://github.com/apache/hadoop/pull/6468#issuecomment-1926348440)

I am not sure why would a different process call abortTask, The driver process should ideally call abortJob if a job fails.

Reply:
looked at the hadoop stuff and yes, it does it in the same process too
```
this is called from a task's process to clean 
up a single task's output that can not yet been committed. This may be
called multiple times for the same task, but for different task attempts.
```
so nothing to worry about